OOModel/src/expressions/MetaCallExpression.cpp

```diff
@@ -62,4 +64,30 @@ MetaDefinition* MetaCallExpression::metaDefinition()
 	return ret;
 }
 
+Declaration* MetaCallExpression::generate()
+{
+	auto metaDef = metaDefinition();
+	if (!metaDef) return nullptr;
+
+	// every formal argument must have a value
+	if (arguments()->size() != metaDef->arguments()->size())
+	{
+		qDebug() << "#metaDefArgs != #metaCallArgs";
+		return nullptr;
+	}
+
+	QMap<QString, Model::Node*> args;
+	for (auto i = 0; i < arguments()->size(); i++)
+	{
+		args.insert(metaDef->arguments()->at(i)->name(), arguments()->at(i));
+	}
+
+	auto cloned = metaDef->context()->clone();
+
+	CodeGenerationVisitor codeGenVisitor (args);
+	codeGenVisitor.visit(cloned);
```

Comment:
For now I only need to clone the context here. But clone is used at other places too. Yes, this is why I applied the hack.